replica/table.cc

```diff
@@ -688,6 +688,9 @@ table::try_flush_memtable_to_sstable(lw_shared_ptr<memtable> old, sstable_write_
                 _memtables->erase(old);
                 co_return stop_iteration::yes;
             }
+            if (!_async_gate.is_closed() || is_auto_compaction_disabled_by_user()) {
```

Comment:
Shouldn't this be `!is_auto_compaction_disabled_by_user()`?

Reply:
almost... :)
It should '&& !is_auto_compaction_disabled_by_user' (or !(_async_gate.is_closed() || is_auto_compaction_disabled_by_user()) which was the original intention)

good catch

Reply:
I'll just delete this term, since it's checked again anyhow by `maybe_wait_for_sstable_count_reduction`